src/rez/pip.py

```diff
@@ -202,6 +202,9 @@ def find_pip_from_context(python_version, pip_version=None):
 
     py_exe = find_python_in_context(context)
 
+    if not py_exe:
+        raise RezSystemError("Failed to locate Python executable from context {}".format(context))
```

Comment:
I think returning early like `return None, None, context` would be slightly better. That's because of the way the function is called: https://github.com/nerdvegas/rez/blob/master/src/rez/pip.py#L92-L101.